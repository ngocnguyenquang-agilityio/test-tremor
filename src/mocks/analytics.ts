export const ANALYTIC_SALES_CARD = [
  {
    flag: "/assets/images/flags/US.webp",
    country: "United State",
    sales: "2500",
    value: "$230,900",
    bounce: "29.9%",
  },
  {
    flag: "/assets/images/flags/DE.webp",
    country: "Germany",
    sales: "3.900",
    value: "$440,000",
    bounce: "40.22%",
  },
  {
    flag: "/assets/images/flags/GB.webp",
    country: "Great Britain",
    sales: "1.400",
    value: "$190,700",
    bounce: "23.44%",
  },
  {
    flag: "/assets/images/flags/BR.webp",
    country: "Brasil",
    sales: "562",
    value: "$143,960",
    bounce: "32.14%",
  },
];

export const STATISTICAL_DATA = [
  {
    type: "Bookings",
    amount: "281",
    totalAmount: "55%",
    duration: "than last week",
  },
  {
    type: "Today's Users",
    amount: "2,300",
    totalAmount: "3%",
    duration: "than last month",
  },
  {
    type: "Revenue",
    amount: "34k",
    totalAmount: "1%",
    duration: "than yesterday",
  },
  {
    type: "Followers",
    amount: "+91",
    totalAmount: "",
    duration: "Just updated",
  },
];
