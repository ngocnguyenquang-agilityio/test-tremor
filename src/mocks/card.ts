export const PROFILE_INFO = {
  description:
    "Hi, I’m Alec Thompson, Decisions: If you can’t decide, the answer is no. If two equally difficult paths, choose the one more painful in the short term (pain avoidance is creating an illusion of equality)",
  info: [
    {
      key: "Full Name",
      value: "Alec M. Thompson",
    },
    {
      key: "Mobile",
      value: "(44) 123 1234 123",
    },
    {
      key: "Email",
      value: "alecthompson@mail.com",
    },
    {
      key: "Location",
      value: "USA",
    },
  ],
};
