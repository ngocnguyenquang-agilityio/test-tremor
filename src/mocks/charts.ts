export const LINE_CHART_DATA = [
  {
    id: "sale",
    display: "Daily Sales",
    descValue: "(+15%)",
    desc: "increase in today sales.",
    modified: "updated 4 min ago",
    data: [
      {
        date: "Apr",
        "Desktop apps": 50,
      },
      {
        date: "May",
        "Desktop apps": 40,
      },
      {
        date: "Jun",
        "Desktop apps": 300,
      },
      {
        date: "Jul",
        "Desktop apps": 220,
      },
      {
        date: "Aug",
        "Desktop apps": 500,
      },
      {
        date: "Sep",
        "Desktop apps": 250,
      },
      {
        date: "Oct",
        "Desktop apps": 400,
      },
      {
        date: "Nov",
        "Desktop apps": 230,
      },
      {
        date: "Dec",
        "Desktop apps": 500,
      },
    ],
  },
  {
    id: "performance",
    display: "Completed Tasks",
    descValue: "(+15%)",
    desc: "Last Campaign Performance",
    modified: "just updated",
    data: [
      {
        date: "Apr",
        "Mobile apps": 50,
      },
      {
        date: "May",
        "Mobile apps": 40,
      },
      {
        date: "Jun",
        "Mobile apps": 300,
      },
      {
        date: "Jul",
        "Mobile apps": 320,
      },
      {
        date: "Aug",
        "Mobile apps": 500,
      },
      {
        date: "Sep",
        "Mobile apps": 350,
      },
      {
        date: "Oct",
        "Mobile apps": 200,
      },
      {
        date: "Nov",
        "Mobile apps": 230,
      },
      {
        date: "Dec",
        "Mobile apps": 500,
      },
    ],
  },
];
