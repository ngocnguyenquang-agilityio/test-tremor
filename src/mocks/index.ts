export * from "./card";
export * from "./profile";
export * from "./project";
export * from "./charts";
export * from "./analytics";
