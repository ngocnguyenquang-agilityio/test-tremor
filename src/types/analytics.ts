export interface IAnalyticsInfo {
  id: string;
  photo: string;
  name: string;
  description: string;
  price: number | string;
  location: string;
}
