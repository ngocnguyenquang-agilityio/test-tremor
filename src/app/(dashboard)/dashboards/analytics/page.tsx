// Components
import { Flex } from "@tremor/react";
import AnalyticsSalesCard from "@/components/AnalyticsSalesCard/AnalyticsSalesCard";
import AnalyticsInfo from "@/components/AnalyticsInfo/AnalyticsInfo";
import AnalyticsStatisticCard from "@/components/AnalyticsStatisticCard/AnalyticsStatisticCard";
import ColumnChart from "@/components/ColumnChart/ColumnChart";
import AnalyticsLineChart from "@/components/AnalyticsLineChart/AnalyticsLineChart";

//Types
import { IAnalyticsInfo, CHART_TYPE } from "@/types";

type AnalyticsStatistical = {
  type: string;
  amount: string;
  totalAmount: string;
  duration: string;
};

// Mock data
import {
  ANALYTIC_SALES_CARD,
  STATISTICAL_DATA,
  ANALYTIC_INFO,
  WEBSITE_CHART,
} from "@/mocks/analytics";
import { ANALYTICS_DAILY_CHART, ANALYTICS_TASK_CHART } from "@/mocks/charts";

const Analytics = () => {
  return (
    <Flex className="flex-col flex-wrap justify-start">
      {/* Sales card  */}
      <AnalyticsSalesCard
        title="Sales by Country"
        chart="/assets/images/analytics/analytics-sales-chart.webp"
        isAnalytics={true}
        data={ANALYTIC_SALES_CARD}
      />
      {/* Charts */}
      <Flex className="mt-12 grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-12 md:gap-x-6">
        {/* Website Views chart */}
        <Flex>
          <ColumnChart
            data={WEBSITE_CHART}
            title={"website views"}
            subTitle={"Last Campaign Performance"}
            scheduleText={"campaign sent 2 days ago"}
          />
        </Flex>
        {/* Daily Sales chart */}
        <Flex>
          <AnalyticsLineChart
            dataChart={ANALYTICS_TASK_CHART}
            type={CHART_TYPE.TASK}
            title={"Daily Sales"}
            subTitle={"increase in today sales."}
            scheduleText={"updated 4 min ago"}
            isDailyChart={true}
          />
        </Flex>
        {/* Completed Tasks chart */}
        <Flex>
          <AnalyticsLineChart
            dataChart={ANALYTICS_DAILY_CHART}
            title={"Completed Tasks"}
            subTitle={"Last Campaign Performance"}
            scheduleText={"just updated"}
            isDailyChart={false}
          />
        </Flex>
      </Flex>
      {/* Statistic cards  */}
      <Flex className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-y-0 lg:gap-y-6 md:gap-x-6">
        {STATISTICAL_DATA?.map((item: AnalyticsStatistical) => (
          <AnalyticsStatisticCard key={item.type} statisticalData={item} />
        ))}
      </Flex>
      {/* Info cards  */}
      <Flex className="justify-start flex-wrap lg:flex-nowrap flex-col md:flex-row items-start mt-12">
        {ANALYTIC_INFO?.map((item: IAnalyticsInfo) => (
          <AnalyticsInfo key={item.id} infoData={item} />
        ))}
      </Flex>
    </Flex>
  );
};

export default Analytics;
