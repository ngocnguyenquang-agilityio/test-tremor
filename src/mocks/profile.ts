import { ConversationHistory } from "@/types/profile";

export const PROFILE_HEADER = {
  avatarUrl: "/images/avatar/avatar-lg.webp",
  alt: "image",
  name: "Richard Davis",
  role: "CEO / Co-Founder",
};

export const PROFILE_ITEM = {
  id: 1,
  avatar: "/images/avatar/avatar-md.webp",
  name: "Sophie.B",
  lastConversation: "Hi! I need more information..",
};

export const PROFILE_CONVERSATIONS: ConversationHistory[] = [
  {
    id: 1,
    avatar: "/images/avatar/avatar-md.webp",
    name: "Anne Marie",
    lastConversation: "Awesome work, can you..",
  },
  {
    id: 2,
    avatar: "/images/avatar/avatar-lg.webp",
    name: "Ivanna",
    lastConversation: "About files I can..",
  },
  {
    id: 3,
    avatar: "/images/avatar/avatar-xs.webp",
    name: "Peterson",
    lastConversation: "Have a great afternoon..",
  },
  {
    id: 4,
    avatar: "/images/avatar/avatar-md.webp",
    name: "Sophie.B",
    lastConversation: "Hi! I need more information..",
  },
  {
    id: 5,
    avatar: "/images/avatar/avatar-sm.webp",
    name: "Nick Daniel",
    lastConversation: "Hi! I need more information..",
  },
];

export const PROFILE_INFO_PROJECT_CARD = [
  {
    id: 1,
    cover: "/assets/images/profile-project.webp",
    participants: [
      { avatar: "/images/avatar/avatar-xs.webp" },
      { avatar: "/images/avatar/avatar-xs.webp" },
    ],
    primaryName: "modern",
    description:
      "As Uber works through a huge amount of internal management turmoil.",
    secondaryName: "#1",
  },
  {
    id: 2,
    cover: "/assets/images/profile-project.webp",
    participants: [
      { avatar: "/images/avatar/avatar-xs.webp" },
      { avatar: "/images/avatar/avatar-xs.webp" },
    ],
    primaryName: "scandinavian",
    description:
      "As Uber works through a huge amount of internal management turmoil.",
    secondaryName: "#2",
  },
  {
    id: 3,
    cover: "/assets/images/profile-project.webp",
    participants: [
      { avatar: "/images/avatar/avatar-xs.webp" },
      { avatar: "/images/avatar/avatar-xs.webp" },
    ],
    primaryName: "minimalist",
    description:
      "Different people have different taste, and various types of music.",
    secondaryName: "#2",
  },
  {
    id: 4,
    cover: "/assets/images/profile-project.webp",
    participants: [
      { avatar: "/images/avatar/avatar-xs.webp" },
      { avatar: "/images/avatar/avatar-xs.webp" },
    ],
    primaryName: "gothic",
    description:
      "Why would anyone pick blue over pink? Pink is obviously a better color.",
    secondaryName: "#4",
  },
];
