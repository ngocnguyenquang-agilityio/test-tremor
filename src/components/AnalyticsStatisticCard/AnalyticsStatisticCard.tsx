//Libs
import { ReactNode } from "react";

//Components
import { Card, Text, Flex } from "@tremor/react";
import { MdLanguage } from "react-icons/md";
import IconBox from "@/components/IconBox/IconBox";

//Mocks
import { STATISTICAL_DATA } from "@/mocks/analytics";

type AnalyticsStatistical = {
  type: string;
  amount: string;
  totalAmount: string;
  duration: string;
};

interface IAnalyticsStatisticCard {
  statisticalData: AnalyticsStatistical;
  bgIcon?: string;
  icon?: ReactNode;
}

const AnalyticsStatisticCard = ({
  statisticalData = STATISTICAL_DATA[0],
  bgIcon = "bg-[linear-gradient(195deg,#42424a,#191919)]",
  icon = <MdLanguage color="white" size="24px" />,
}: IAnalyticsStatisticCard): JSX.Element => {
  const { type, amount, totalAmount, duration } = statisticalData;

  return (
    <div className="font-primary antialiased items-center justify-between py-1">
      <div className="flex items-center">
        <Card className="mx-auto py-3 px-4 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border-none relative mt-[40px] rounded-xl shadow-md">
          <Flex className="">
            <IconBox
              className="absolute top-[-22px]"
              bgBox={bgIcon}
              icon={icon}
            />
            <Flex className="pl-[90px] flex-col mb-1 items-end">
              <Text className="text-md text-secondary font-light">{type}</Text>
              <Text className="!text-2xl text-primary leading-[33px] tracking-[0.1764px] mt-3 font-bold">
                {amount}
              </Text>
            </Flex>
          </Flex>
          <div className="h-px bg-[linear-gradient(to_right,rgba(52,71,103,0),rgba(52,71,103,0.4),rgba(52,71,103,0))] opacity-25 my-4" />
          <Flex>
            <Flex className="justify-start items-start tracking-[0.4px]">
              <Text className="text-few leading-[21px] font-bold">
                {totalAmount}
              </Text>
              <Text className="ml-1 text-secondary leading-[21px] font-light">
                {duration}
              </Text>
            </Flex>
          </Flex>
        </Card>
      </div>
    </div>
  );
};

export default AnalyticsStatisticCard;
