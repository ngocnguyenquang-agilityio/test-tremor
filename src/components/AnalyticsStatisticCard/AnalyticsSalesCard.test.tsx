import { render } from "@testing-library/react";

//Components
import AnalyticsStatisticCard from "./AnalyticsStatisticCard";

//Mocks
import { STATISTICAL_DATA } from "@/mocks/analytics";

describe("Testing AnalyticsStatisticCard component", () => {
  const propsDefault = {
    statisticalData: STATISTICAL_DATA[0],
  };

  // TODO: Will update soon
  it.skip("Should match snapshot", () => {
    const component = render(<AnalyticsStatisticCard {...propsDefault} />);
    expect(component).toMatchSnapshot();
  });
});
