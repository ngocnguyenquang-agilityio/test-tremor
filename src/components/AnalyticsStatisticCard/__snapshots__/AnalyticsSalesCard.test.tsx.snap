// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AnalyticsStatisticCard component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="font-primary antialiased items-center justify-between py-1"
      >
        <div
          class="flex items-center"
        >
          <div
            class="tremor-Card-root w-full text-left rounded-tremor-default bg-tremor-background ring-tremor-ring shadow-tremor-card dark:bg-dark-tremor-background dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand p-6 mx-auto py-3 px-4 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border-none relative mt-[40px] rounded-xl shadow-md"
          >
            <div
              class="tremor-Flex-root flex w-full flex-row justify-between items-center"
            >
              <div
                class="tremor-Flex-root flex flex-row items-center p-1 justify-center rounded-xl w-16 h-16 bg-[linear-gradient(195deg,#42424a,#191919)] absolute top-[-22px] left-40px"
              >
                <svg
                  color="white"
                  fill="currentColor"
                  height="24px"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: white;"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95a15.65 15.65 0 00-1.38-3.56A8.03 8.03 0 0118.92 8zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56A7.987 7.987 0 015.08 16zm2.95-8H5.08a7.987 7.987 0 014.33-3.56A15.65 15.65 0 008.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95a8.03 8.03 0 01-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
                  />
                </svg>
              </div>
              <div
                class="tremor-Flex-root flex w-full justify-between pl-[90px] flex-col mb-1 items-end"
              >
                <p
                  class="text-tremor-default dark:text-dark-tremor-content text-md text-[#7b809a] font-light"
                >
                  Bookings
                </p>
                <p
                  class="text-tremor-default dark:text-dark-tremor-content text-[#344767] text-[24px] mt-3 font-bold"
                >
                  281
                </p>
              </div>
            </div>
            <div
              class="h-px bg-[linear-gradient(to_right,rgba(52,71,103,0),rgba(52,71,103,0.4),rgba(52,71,103,0))] opacity-25 my-4"
            />
            <div
              class="tremor-Flex-root flex w-full flex-row justify-between items-center"
            >
              <div
                class="tremor-Flex-root flex w-full flex-row justify-start items-start"
              >
                <p
                  class="text-tremor-default dark:text-dark-tremor-content text-[#4caf50] font-bold"
                >
                  55%
                </p>
                <p
                  class="text-tremor-default dark:text-dark-tremor-content ml-1 text-[#7b809a] font-light"
                >
                  than last week
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="font-primary antialiased items-center justify-between py-1"
    >
      <div
        class="flex items-center"
      >
        <div
          class="tremor-Card-root w-full text-left rounded-tremor-default bg-tremor-background ring-tremor-ring shadow-tremor-card dark:bg-dark-tremor-background dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand p-6 mx-auto py-3 px-4 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border-none relative mt-[40px] rounded-xl shadow-md"
        >
          <div
            class="tremor-Flex-root flex w-full flex-row justify-between items-center"
          >
            <div
              class="tremor-Flex-root flex flex-row items-center p-1 justify-center rounded-xl w-16 h-16 bg-[linear-gradient(195deg,#42424a,#191919)] absolute top-[-22px] left-40px"
            >
              <svg
                color="white"
                fill="currentColor"
                height="24px"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95a15.65 15.65 0 00-1.38-3.56A8.03 8.03 0 0118.92 8zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56A7.987 7.987 0 015.08 16zm2.95-8H5.08a7.987 7.987 0 014.33-3.56A15.65 15.65 0 008.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95a8.03 8.03 0 01-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
                />
              </svg>
            </div>
            <div
              class="tremor-Flex-root flex w-full justify-between pl-[90px] flex-col mb-1 items-end"
            >
              <p
                class="text-tremor-default dark:text-dark-tremor-content text-md text-[#7b809a] font-light"
              >
                Bookings
              </p>
              <p
                class="text-tremor-default dark:text-dark-tremor-content text-[#344767] text-[24px] mt-3 font-bold"
              >
                281
              </p>
            </div>
          </div>
          <div
            class="h-px bg-[linear-gradient(to_right,rgba(52,71,103,0),rgba(52,71,103,0.4),rgba(52,71,103,0))] opacity-25 my-4"
          />
          <div
            class="tremor-Flex-root flex w-full flex-row justify-between items-center"
          >
            <div
              class="tremor-Flex-root flex w-full flex-row justify-start items-start"
            >
              <p
                class="text-tremor-default dark:text-dark-tremor-content text-[#4caf50] font-bold"
              >
                55%
              </p>
              <p
                class="text-tremor-default dark:text-dark-tremor-content ml-1 text-[#7b809a] font-light"
              >
                than last week
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
